imports/ui/components/dashboard/match_card.jsx

```diff
@@ -1,49 +1,123 @@
   constructor(props) {
     super(props);
+    this.showFlagModal = this.showFlagModal.bind(this);
+    this.closeFlagModal = this.closeFlagModal.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.submitFlagModal = this.submitFlagModal.bind(this);
+    this.state = {
+      showFlagModal: false,
+      flagReason: '',
+    };
+  }
+  showFlagModal(event) {
+    event.preventDefault();
+    this.setState({ showFlagModal: true });
+  }
+  closeFlagModal(event) {
+    event.preventDefault();
+    this.setState({ showFlagModal: false });
+  }
+  handleChange(e) {
+    this.setState({
+      flagReason: e.target.value,
+    });
+  }
+  submitFlagModal(e) {
+    e.preventDefault();
+
+    const data = {
+      userId: Meteor.userId(),
+      reason: this.state.flagReason,
+      entryId: this.props.entryId,
+    };
+
+    this.setState({ showFlagModal: false });
+    Meteor.call('entry.flag', data, (error, result) => {
+      if (error) {
+        this.setState({ error: error.reason });
+        this.setState({ processing: false });
+      }
+      if (result) {
+        this.setState({ processing: false });
+      }
+    });
   }
 
   render() {
     const { oneLineIntro, lookingFor, ownCard, timezone } = this.props;
     return (
-      <Card>
-        <Card.Body>
-          <Card.Title className="font-weight-normal">{oneLineIntro}</Card.Title>
-          <Card.Text>{lookingFor}</Card.Text>
-          {ownCard === 'true' ? (
-            ''
-          ) : (
-            <Dropdown as={ButtonGroup} className="btn-block">
-              <Button className="btn-block" variant="primary">
-                Request Match
-              </Button>
+      <div>
+        <Card>
+          <Card.Body>
+            <Card.Title className="font-weight-normal">{oneLineIntro}</Card.Title>
+            <Card.Text>{lookingFor}</Card.Text>
+            {ownCard === 'true' ? (
+              ''
+            ) : (
+              <Dropdown as={ButtonGroup} className="btn-block">
+                <Button className="btn-block" variant="primary">
+                  Request Match
+                </Button>
 
-              <Dropdown.Toggle split variant="primary" />
-
-              <Dropdown.Menu>
-                <Dropdown.Item as="button">Report</Dropdown.Item>
-              </Dropdown.Menu>
-            </Dropdown>
-          )}
-        </Card.Body>
-        <Card.Footer>
-          <small className="text-muted">{timezone}</small>
-        </Card.Footer>
-      </Card>
+                <Dropdown.Toggle split variant="primary" />
+
+                <Dropdown.Menu>
+                  <Dropdown.Item as="button" onClick={this.showFlagModal}>
+                    Flag
+                  </Dropdown.Item>
+                </Dropdown.Menu>
+              </Dropdown>
+            )}
+          </Card.Body>
+          <Card.Footer>
+            <small className="text-muted">{timezone}</small>
+          </Card.Footer>
+        </Card>
+
```

Comment:
Would it be cleaner to move the modal as its own component to simplify this render method?

Reply:
Oh, great idea!

Reply:
Actually, if you're up for it, let's pair on this sometime! Ran into a few questions that could be more easily addressed if we were looking at it together.